File: go_algorithms/utility/algorithmExecutor.go
package utility

import (
	"fmt"
	"go_algorithms/algorithms"
	"go_algorithms/database"
)

func Init_execution(matrixA [][]int, matrixB [][]int, test string) {
	isPrint := true

	result, time := algorithms.NaiveOnArray(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "NaiveOnArray", time)
	print(time, result, isPrint)

	result, time = algorithms.NaiveLoopUnrollingTwo(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "NaiveLoopUnrollingTwo", time)
	print(time, result, isPrint)

	result, time = algorithms.NaivLoopUnrollingFour(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "NaivLoopUnrollingFour", time)
	print(time, result, isPrint)

	result, time = algorithms.WinogradOriginal(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "WinogradOriginal", time)
	print(time, result, isPrint)

	result, time = algorithms.WinogradScaled(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "WinogradScaled", time)
	print(time, result, isPrint)

	result, time = algorithms.StrassenNaiv(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "StrassenNaiv", time)
	print(time, result, isPrint)

	result, time = algorithms.StrassenWinograd(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "StrassenWinograd", time)
	print(time, result, isPrint)

	result, time = algorithms.RowByColumnSequential(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "RowByColumnSequential", time)
	print(time, result, isPrint)

	result, time = algorithms.RowByColumnParallel(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "RowByColumnParallel", time)
	print(time, result, isPrint)

	result, time = algorithms.RowByColumnEnhanced(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "RowByColumnEnhanced", time)
	print(time, result, isPrint)

	result, time = algorithms.RowByRowSequential(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "RowByRowSequential", time)
	print(time, result, isPrint)

	result, time = algorithms.RowByRowParallel(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "RowByRowParallel", time)
	print(time, result, isPrint)

	result, time = algorithms.RowByColumnEnhanced(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "RowByColumnEnhanced", time)
	print(time, result, isPrint)

	result, time = algorithms.ColumnByColumnSequential(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "ColumnByColumnSequential", time)
	print(time, result, isPrint)

	result, time = algorithms.ColumnByColumnParallel(matrixA, matrixB)
	database.RegisterInfo(test, "Golang", "ColumnByColumnParallel", time)
	print(time, result, isPrint)

	fmt.Print("All algorithms executed correctly\n\n")
}

func print(time int64, matrix [][]int, print bool) {
	if print {
		for i := 0; i < 5 && i < len(matrix); i++ {
			fmt.Printf("%d ", matrix[i][0])
		}
	} else {
		totalValues := len(matrix) * len(matrix[0])
		fmt.Printf(" + %d other values\n = %dns", totalValues-5, time)
	}
}

File: python_algorithms/utility/algorithmExecutor.py
import algorithms.ColumnByColumnParallel 
import algorithms.ColumnByColumnSequential 
import algorithms.NaivLoopUnrollingFour 
import algorithms.NaivLoopUnrollingTwo 
import algorithms.NaivOnArray 
import algorithms.RowByColumnEnhanced 
import algorithms.RowByColumnParallel 
import algorithms.RowByColumnSequential 
import algorithms.RowByRowEnhanced 
import algorithms.RowByRowParallel 
import algorithms.RowByRowSequential 
import algorithms.StrassenNaiv 
import algorithms.StrassenWinograd 
import algorithms.WinogradOriginal 
import algorithms.WinogradScaled
from database.connection import register_info

def init_execution(matrizA, matrizB, test):
    
    result, time = algorithms.NaivOnArray.Multiply(matrizA, matrizB)
    register_info(test, "Python", "NaivOnArray", time)
    print(time, "\n\n")

    result, time = algorithms.NaivLoopUnrollingTwo.Multiply(matrizA, matrizB)
    register_info(test, "Python", "NaivLoopUnrollingTwo", time)
    print(time, "\n\n")

    result, time = algorithms.NaivLoopUnrollingFour.Multiply(matrizA, matrizB)
    register_info(test, "Python", "NaivLoopUnrollingFour", time)
    print(time, "\n\n")

    result, time = algorithms.WinogradOriginal.Multiply(matrizA, matrizB)
    register_info(test, "Python", "WinogradOriginal", time)
    print(time, "\n\n")

    result, time = algorithms.WinogradScaled.Multiply(matrizA, matrizB)
    register_info(test, "Python", "WinogradScaled", time)
    print(time, "\n\n")

    result, time = algorithms.StrassenNaiv.Multiply(matrizA, matrizB)
    register_info(test, "Python", "StrassenNaiv", time)
    print(time, "\n\n")

    result, time = algorithms.StrassenWinograd.Multiply(matrizA, matrizB)
    register_info(test, "Python", "StrassenWinograd", time)
    print(time, "\n\n")

    result, time = algorithms.RowByColumnSequential.Multiply(matrizA, matrizB)
    register_info(test, "Python", "RowByColumnSequential", time)
    print(time, "\n\n")

    result, time = algorithms.RowByColumnParallel.Multiply(matrizA, matrizB)
    register_info(test, "Python", "RowByColumnParallel", time)
    print(time, "\n\n")

    result, time = algorithms.RowByColumnEnhanced.Multiply(matrizA, matrizB)
    register_info(test, "Python", "RowByColumnEnhanced", time)
    print(time, "\n\n")

    result, time = algorithms.RowByRowSequential.Multiply(matrizA, matrizB)
    register_info(test, "Python", "RowByRowSequential", time)
    print(time, "\n\n")

    result, time = algorithms.RowByRowParallel.Multiply(matrizA, matrizB)
    register_info(test, "Python", "RowByRowParallel", time)
    print(time, "\n\n")

    result, time = algorithms.RowByRowEnhanced.Multiply(matrizA, matrizB)
    register_info(test, "Python", "RowByRowEnhanced", time)
    print(time, "\n\n")

    result, time = algorithms.ColumnByColumnSequential.Multiply(matrizA, matrizB)
    register_info(test, "Python", "ColumnByColumnSequential", time)
    print(time, "\n\n")

    result, time = algorithms.ColumnByColumnParallel.Multiply(matrizA, matrizB)
    register_info(test, "Python", "ColumnByColumnParallel", time)
    print(time, "\n\n")

